sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -632,9 +632,51 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     val rawTable = getRawTable(db, table)
     val withNewSchema = rawTable.copy(schema = schema)
     verifyColumnNames(withNewSchema)
+
+    val newProperties = collection.mutable.Map[String, String]()
+    for ((property, value) <- withNewSchema.properties) {
+      newProperties.put(property, value)
+    }
+    for ((property, value) <- tableMetaToTableProps(withNewSchema)) {
+      newProperties.put(property, value)
+    }
+
+    // Since the table schema will be changed, check if the bucketing and sort columns are still
```

Comment:
I have to do this because one of the unit tests was failing : https://github.com/apache/spark/blob/703c42c398fefd3f7f60e1c503c4df50251f8dcf/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogSuite.scala#L246

In that test case, the table has a bucketing and alter removes the bucketing column. The way I interpret things: if the alter command is not removing the bucketing / sorting columns, we should keep the original bucketing spec. If any one of the bucketing / sorting columns are removed in alter, then the bucketing spec needs to be reset to empty (else we might refer to dangling columns).